node/nodeTransactions_test.go

```diff
@@ -102,13 +118,166 @@ func TestNode_GetTransaction_ShouldFindInTxStorageAndReturn(t *testing.T) {
 		node.WithInternalMarshalizer(&mock.MarshalizerFake{}, 0),
 		node.WithAddressPubkeyConverter(&mock.PubkeyConverterMock{}),
 		node.WithShardCoordinator(&mock.ShardCoordinatorMock{}),
+		node.WithHistoryProcessor(&mock.HistoryProcessorStub{
+			IsEnabledCalled: func() bool {
+				return false
+			},
+		}),
 	)
 	expectedTx, _ := getDummyNormalTx()
 	tx, err := n.GetTransaction("aaaa")
 	assert.NoError(t, err)
 	assert.Equal(t, expectedTx.Nonce, tx.Nonce)
 }
 
+func TestNode_GetFullHistoryTransaction(t *testing.T) {
+	t.Parallel()
+
+	dataPool := &testscommon.PoolsHolderStub{
+		TransactionsCalled:         getCacherHandler(false, ""),
+		RewardTransactionsCalled:   getCacherHandler(false, ""),
+		UnsignedTransactionsCalled: getCacherHandler(false, ""),
+	}
+	storer := &mock.ChainStorerMock{
+		GetStorerCalled: func(unitType dataRetriever.UnitType) storage.Storer {
+			return getStorerStub(true)
+		},
+	}
+
+	blockHash := []byte("hash")
+	mbHash := []byte("mbHash")
+	epoch := uint32(10)
+	sndShard := uint32(1)
+	rcvShard := uint32(2)
+	round := uint64(123)
+	blockNonce := uint64(1001)
+	n, _ := node.NewNode(
+		node.WithDataPool(dataPool),
+		node.WithDataStore(storer),
+		node.WithInternalMarshalizer(&mock.MarshalizerFake{}, 0),
+		node.WithAddressPubkeyConverter(&mock.PubkeyConverterMock{}),
+		node.WithShardCoordinator(&mock.ShardCoordinatorMock{}),
+		node.WithHistoryProcessor(&mock.HistoryProcessorStub{
+			IsEnabledCalled: func() bool {
+				return true
+			},
+			GetTransactionCalled: func(hash []byte) (*fullHistory.HistoryTransactionWithEpoch, error) {
+				return &fullHistory.HistoryTransactionWithEpoch{
+					Epoch: epoch,
+					HistoryTransaction: &fullHistory.HistoryTransaction{
+						MbHash:      mbHash,
+						HeaderHash:  blockHash,
+						HeaderNonce: blockNonce,
+						SndShardID:  sndShard,
+						RcvShardID:  rcvShard,
+						Round:       round,
+					},
+				}, nil
+			},
+		}),
+	)
+
+	dummyTx, _ := getDummyNormalTx()
+	expectedTx := &transaction.ApiTransactionResult{
```

Comment:
Not visible why `expectedTx` is expected - perhaps describe the test in some comments?

Reply:
added comments